dali/operators/decoder/video/video_decoder_base.h

```diff
@@ -17,52 +17,38 @@
 
 #include "dali/pipeline/operator/common.h"
 #include "dali/pipeline/operator/operator.h"
-#include "dali/operators/reader/loader/video/frames_decoder.h"
-#include "dali/operators/reader/loader/video/frames_decoder_gpu.h"
 
-#ifndef DALI_OPERATORS_DECODER_VIDEO_VIDEO_DECODER_H_
-#define DALI_OPERATORS_DECODER_VIDEO_VIDEO_DECODER_H_
+#ifndef DALI_OPERATORS_DECODER_VIDEO_VIDEO_DECODER_BASE_H_
+#define DALI_OPERATORS_DECODER_VIDEO_VIDEO_DECODER_BASE_H_
 
 namespace dali {
 
-template <typename Backend>
-struct FramesDecoderImpl {};
-
-template <>
-struct FramesDecoderImpl<CPUBackend> {
-  using type = FramesDecoder;
-};
-
-template <>
-struct FramesDecoderImpl<MixedBackend> {
-  using type = FramesDecoderGpu;
-};
-
-template <typename Backend>
-using FramesDecoder_t = typename FramesDecoderImpl<Backend>::type;
-
-template <typename Backend>
-class DLL_PUBLIC VideoDecoder : public Operator<Backend> {
+template <typename Backend, typename FramesDecoder>
+class DLL_PUBLIC VideoDecoderBase : public Operator<Backend> {
  public:
-  explicit VideoDecoder(const OpSpec &spec)
+  using InBackend = CPUBackend;
+  using OutBackend = std::conditional_t<std::is_same_v<Backend, CPUBackend>,
```

Comment:
```suggestion
  using OutBackend = std::conditional_t<std::is_same_v<Backend, MixedBackend>,
```
As I understand the GPU decoder is going to be a mixed operator, right?